backend/data/processors/load_conditions.py

```diff
@@ -16,21 +19,30 @@ def cache_conditions_pkl_file():
     Input: None
     Returns: None
     """
+    all_objects = construct_conditions_objects()
 
-    # Load in all the courses and their conditions list
-    all_conditions_tokens_file = "./data/final_data/conditionsTokens.json"
-    pickle_file = "./data/final_data/conditions.pkl"
+    # Dump the dictionary as a pickle file
+    with open(CONDITIONS_PICKLE_FILE, "wb") as f_out:
+        pickle.dump(all_objects, f_out, pickle.HIGHEST_PROTOCOL)
 
-    with open(all_conditions_tokens_file, "r", encoding="utf8") as f:
-        all_conditions_tokens = json.load(f)
+def construct_conditions_objects() -> Dict[str, CompositeCondition | None]:
+    """
```

Comment:
Define this function before the one above?
I know that it doesn't actually matter, but it's confusing to have your first function in your file dump stufd

Reply:
👍